program_xdp.go

```diff
@@ -61,27 +78,49 @@ func newXdpProgram(name, license string, bytecode []byte) Program {
 	}
 }
 
+// Attach attaches eBPF(XDP) program to network interface.
+// There are 2 possible ways to do that:
+//
+// 1. Pass interface name as parameter, e.g.
+//    xdpProgram.Attach("eth0")
+//
+// 2. Using XdpAttachParams structure:
+//    xdpProgram.Attach(
+//			&XdpAttachParams{Mode: XdpAttachModeSkb, Interface: "eth0"
+//    })
 func (p *xdpProgram) Attach(data interface{}) error {
-	ifname, ok := data.(string)
-	if !ok {
-		return fmt.Errorf("Interface name as string expected, got %T", data)
+	var ifaceName string
+	var attachMode = XdpAttachModeDrv
```

Comment:
mb default to 0? which means "best effor" if DRV mode is supported - it would be attached in DRV mode. otherwise it would fallback to SKB etc

Reply:
Current version is using `0`, but, it actually does not work as we discovered on DC test machine:

https://github.com/vishvananda/netlink/blob/f049be6f391489d3f374498fe0c8df8449258372/link_linux.go#L830-L832

Anyway changed to zero.

Reply:
0 means it would try DRV first. And only if driver doesn’t not support DRV - fallback to skb. In our usecase driver supports DRV mode but fails to attach because of Number of possible CPU’s.